packages/drawer/rc/__stories__/index.story.js

```diff
@@ -1,262 +1,225 @@
+                    }, eatEvent)}
+                  >
+                    close
+                  </Button>
+                </div>
+              }
+              title="Click should not toggle"
+            />
           </Drawer.Head>
+
           <Drawer.Body>
-            <DrawerBodyContent>Drawer Content here</DrawerBodyContent>
+            <BodyContent />
           </Drawer.Body>
         </Drawer>
-      </div>
+      )
+    }
 
-      <div>
-        <Drawer>
-          <Drawer.Head>
-            <DrawerHeadContent>The Drawer #3</DrawerHeadContent>
-          </Drawer.Head>
-          <Drawer.Body>
-            <DrawerBodyContent>Drawer Content here</DrawerBodyContent>
-          </Drawer.Body>
+    return <Story />
+  })
+  .add('prop: onRequest(Open/Close/)', () => {
+    function Story() {
+      const [isOpen, setIsOpen] = useState(true)
+
+      return (
+        <Drawer
+          isOpen={isOpen}
+          onRequestOpen={() => setIsOpen(true)}
+          onRequestClose={() => setIsOpen(false)}
+        >
+          {({ isOpen, close, open }) => (
+            <>
+              <Drawer.Head onClick={eatEvent}>
+                <HeadContent
+                  actions={
+                    <div>
+                      <Button
+                        disabled={isOpen}
+                        onClick={combineFns(open, eatEvent)}
+                      >
+                        open
+                      </Button>
+
+                      <Button
+                        disabled={!isOpen}
+                        onClick={combineFns(close, eatEvent)}
+                      >
+                        close
+                      </Button>
+                    </div>
+                  }
+                />
+              </Drawer.Head>
+
+              <Drawer.Body>
+                <BodyContent />
+              </Drawer.Body>
+            </>
+          )}
         </Drawer>
-      </div>
-    </>
-  )) // Is accessibility labels something we should be responsible for
-  .add('using custom aria label', () => (
-    <Drawer>
-      <Drawer.Head aria-label="custom drawer header">
-        <DrawerHeadContent>The Drawer</DrawerHeadContent>
-      </Drawer.Head>
-      <Drawer.Body aria-label="custom drawer body">
-        <DrawerBodyContent>Drawer Content here</DrawerBodyContent>
-      </Drawer.Body>
-    </Drawer>
-  ))
+      )
+    }
+
+    return <Story />
+  })
+
+storiesOf('Drawer | RC / advanced', module)
+  .addDecorator(story => <div style={{ padding: 48 }}>{story()}</div>)
+  .add('default', () => {
+    function Story() {
+      const initialState = { isOpen: true }
+      const stateReducer = (prevState, action, pendingChanges) => {
+        if (action.type === Drawer.types.close && !confirm('allow to close'))
+          return prevState
+
+        if (action.type === Drawer.types.open && !confirm('allow to open'))
+          return prevState
+
+        return pendingChanges
+      }
```

Comment:
the internal state/actions/types are now part of the public api so we'll have to ensure we mark breaking changes appropriately. might not be that big of a deal but i thought i'd point it out